gui/wxpython/startup/utils.py

```diff
@@ -87,3 +90,77 @@ def create_database_directory():
         pass
 
     return None
+
+
+def _get_startup_location_in_distribution():
+    """Check for startup location directory in distribution.
+
+    Returns startup location if found or None if nothing was found.
+    """
+    gisbase = os.getenv("GISBASE")
+    startup_location = os.path.join(gisbase, "demolocation")
+
+    # Find out if startup location exists
+    if os.path.exists(startup_location):
+        return startup_location
+    return None
+
+
+def _copy_startup_location(grassdatabase, startup_location):
+    """Copy the simple startup_location with some data to GRASS database.
+
+    Returns path to startup location if successfully copied or None
+    when an error was encountered.
+    """
+    src = startup_location
+    dst = os.path.join(grassdatabase, "world_latlong_wgs84")
+
+    # Copy source startup location into GRASS database
+    try:
+        copytree(src, dst, ignore=ignore_patterns('*.tmpl','Makefile*'))
+        return dst
+    except (IOError, OSError):
+        pass
+    return None
+
+
+def _create_startup_mapset(location_path):
+    """Create the new empty startup mapset named after user.
+
+    Returns True if successfully created or False when an error was encountered.
+    """
+    name = get_default_mapset_name()
+    mapset_path = os.path.join(location_path, name)
+    counter = 2
+
+    while os.path.exists(mapset_path):
+        mapset_name = _("{name}_{counter}").format(name=name,
+                                                   counter=str(counter))
+        mapset_path = os.path.join(location_path, mapset_name)
+        counter += 1
+
+    # Create new startup mapset
+    try:
+        grassdatabase, location = os.path.split(location_path)
+        create_mapset(grassdatabase, location, mapset_name)
+        return grassdatabase, location, mapset_name
+    except (IOError, OSError):
+        pass
+    return None
+
+
+def get_startup_location(grassdatabase):
+    """Wrapping function for managing startup location.
+
+    Returns the copied location on success and None otherwise.
+    Internally resolves all the cases (no location to copy, copying failed
+    and an existing user's copy of the location)."""
```

Comment:
Is the _existing user's copy_ something you already tried to address or is it just something you are planning? I don't see it in the code.

Reply:
I think that the existing user's copy of "world_latlong_wgs84" is something that could not be reached now because I create a startup location only in the case when "grassdata" does not exist. So it is clear that "grassdata/world_latlong_wgs84" does not exist either.